plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts

```diff
@@ -49,12 +52,11 @@ export interface useCollectionOptions {
 	isEditor?: boolean;
 }
 
-export const useCollection = < T >(
+export const useCollection = < T, IsCollection extends boolean = false >(
```

Comment:
At this point, I would incline with a simple type like you did earlier
```suggestion
export const useCollection = < T >(
```

Reply:
I've added the optional `IsCollection` parameter to simplify the type control. Otherwise, `results` could be `T` or `T[]`, triggering TS issues in many other files and what's more important, the ambiguity of the types makes it **unneeded** more complex.
Whit this optional parameter, it's possible to define the unique type of the `results` property.

For instance:

```tsx
const { results: collectionFilters, isLoading: filteredCountsLoading } =
	useCollectionData< WCStoreV1ProductsCollectionProps >( {
		queryRating: true,
		queryState,
		isEditor: true,
	} );
```

Here, `collectionFilters` is an `WCStoreV1ProductsCollectionProps` object
The screenshot below shows how is properly tipified the `collectionFilters` constant:
<img width="1148" alt="image" src="https://github.com/user-attachments/assets/98209129-e27e-48f3-99d2-fc3a1a278b8a">

But also, using the `useCollection()` hook:

```tsx
const { results: products, isLoading: productsLoading } = useCollection<
	ProductResponseItem[]
>( {
	...collectionOptions,
	query,
} );
```

Here, `products` is an array of `ProductResponseItem`. So, for instance, when typing `products[ 0 ].`...

<img width="735" alt="image" src="https://github.com/user-attachments/assets/05e3f6a9-5d8d-4e6c-bacc-5ae1cbb19588">

...typescript recognizes very will the types.

Your suggestion is okay, but a bit ambiguous.